File: 557.reverse-words-in-a-string-iii.js
/*
 * @lc app=leetcode id=557 lang=javascript
 *
 * [557] Reverse Words in a String III
 */

// @lc code=start
/**
 * @param {string} s
 * @return {string}
 */

/* 1. 我自己的方法 */
var reverseWords = function (s) {
    let stringArray = s.split(" ");
    stringArray.forEach((item, index) => {
        stringArray[index] = item.split("").reverse().join("");
    });

    return stringArray.join(" ")
};

/* 2. 最优解 */


reverseWords("Let's take LeetCode contest");

// @lc code=end

File: 557.reverse-words-in-a-string-iii.java
/*
 * @lc app=leetcode id=557 lang=java
 *
 * [557] Reverse Words in a String III
 */

// @lc code=start
class Solution {
    public String reverseWords(String s) {

        String strArray[] = s.split(" ");
        for (int i = 0; i < strArray.length; i++) {
            StringBuilder sb = new StringBuilder(strArray[i]);
            sb = sb.reverse();
            strArray[i] = sb.toString();
        }
        String str = String.join(" ", strArray);
        return str;
    }
}
// @lc code=end
